### DISCUSSION TITLE
Generating ziggy.js uses testing domain

### DISCUSSION THREAD

**Discussion by beard7** [tighten/ziggy#776] [Q&A]
### Ziggy version

2.3.0

### Laravel version

10

### Description

I think this has been covered elsewhere, but I've not found a definitive answer.

When using `php artisan ziggy:generate` the APP_URL is hard-coded into the resulting `ziggy.js` file, so if, like me, you are using Vite and Vue, the resulting assets include the wrong route domain when transferred to the production server.

What's the solution to this? I obviously don't want to build assets on the production server, and I don't want to edit ziggy.js before building assets. Nor do I want to use the @routes Blade directive in my Vue root file.

### Ziggy call and context

```js
n/a
```

### Ziggy configuration

```js
n/a
```

### Route definition

```php
n/a
```

**Comment by bakerkretzmar** [comment]
The `ziggy:generate` command accepts a `--url` argument you can use to override the URL it uses.

**Reply by beard7** [reply]
Okay. I guess that helps a little. It means regenerating the routes with the appropriate URL prior to running vite build for production, and then generating again to continue development. I supppose that's a bit quicker than changing APP_URL in .ENV

**Reply by bakerkretzmar** [reply]
I'd suggest building your production assets as part of your deployment step, either on your production server or in a CI tool like GitHub Actions, then you can run the Ziggy command at the same time with the specific URL for that environment. You might also be able to run it when you deploy and load that file dynamically on your server but that's probably more complicated.
